src/details/ArborX_DetailsHeap.hpp

```diff
@@ -102,7 +105,10 @@ KOKKOS_INLINE_FUNCTION void popHeap(RandomIterator first, RandomIterator last,
     ValueType value = std::move(*first);
     bubbleDown(first, DistanceType(0), DistanceType((last - first) - 1),
                std::move(*(last - 1)), comp);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
     *(last - 1) = std::move(value);
+#pragma GCC diagnostic pop
```

Comment:
This is peculiar. It does not complain on line 106, but is complaining on line 110? And this is despite using the same `*(last - 1)`.